src/main/java/nl/tudelft/ewi/devhub/server/database/controllers/Deliveries.java

```diff
@@ -94,6 +93,32 @@ public boolean lastDeliveryIsApprovedOrDisapproved(Assignment assignment, Group
             .sorted(byState.thenComparing(bySubmissionDate))
             .collect(Collectors.toList());
     }
+    
+    /**
+     * Get the most recent delivery for every group in this assignment
+     * @param assignment current assignment
+     * @return a list of deliveries
+     */
+    @Transactional
+    public List<Delivery> getLastDeliveries(Assignment assignment) {
+    	return this.getLastDeliveries(assignment, null);
+    }
+    
+    /**
+     * Get the most recent delivery for every group in this assignment with the specific state
+     * @param assignment current assignment
+     * @return a list of deliveries
+     */
+    @Transactional
+    public List<Delivery> getLastDeliveries(Assignment assignment, String state) {
+    	BooleanExpression expression = QDelivery.delivery.assignment.eq(assignment);
+
+    	if (state != null) {
```

Comment:
I personally like @LiamClark 's way of doing this more. See the mapper at https://github.com/LiamClark/devhub/commit/2e92648a2749cc2384a43a49290b5751b786769b . Just fetch them all, and group. We're talking about +/- 10 entities anyway.